Source/Core/Core/HW/WiimoteEmu/Encryption.h

```diff
@@ -6,21 +6,52 @@
 
 #pragma once
 
+#include <array>
+
 #include "Common/CommonTypes.h"
 
 namespace WiimoteEmu
 {
 class EncryptionKey
 {
 public:
-  void Generate(const u8* keydata);
+  void Encrypt(u8* data, u32 addr, u32 len) const;
+  void Decrypt(u8* data, u32 addr, u32 len) const;
+
+  void GenerateTables(const std::array<u8, 10>& rand, const std::array<u8, 6>& key, const u8 idx,
+                      const u8 offset);
+
+  std::array<u8, 8> ft = {};
+  std::array<u8, 8> sb = {};
+};
+
+class KeyGen
+{
+public:
+  virtual ~KeyGen() = default;
 
-  void Encrypt(u8* data, int addr, u8 len) const;
-  void Decrypt(u8* data, int addr, u8 len) const;
+  EncryptionKey GenerateFromExtensionKeyData(const std::array<u8, 0x10>& key_data) const;
```

Comment:
It looks a bit weird that this one takes hex 10, while all the other functions use dec 10 (altho it is intentional and not a bug, given that `EncryptedExtension::encryption_key_data` is 16 as well).
Thoughts on using `16` instead? Or a type alias, as mentioned below?

Reply:
:+1: